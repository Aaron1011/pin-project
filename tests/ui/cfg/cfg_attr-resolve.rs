use std::pin::Pin;

#[cfg_attr(any(), pin_project::pin_project)]
struct Foo<T> {
    inner: T,
}

fn foo() {
    let mut x = Foo { inner: 0_u8 };
    let _x = Pin::new(&mut x).project(); //~ ERROR E0599
}

fn main() {}
