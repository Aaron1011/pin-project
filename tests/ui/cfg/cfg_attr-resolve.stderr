error[E0599]: no method named `project` found for type `std::pin::Pin<&mut Foo<u8>>` in the current scope
  --> $DIR/cfg_attr-resolve.rs:10:31
   |
10 |     let _x = Pin::new(&mut x).project();
   |                               ^^^^^^^ method not found in `std::pin::Pin<&mut Foo<u8>>`
